rudder-core/src/main/scala/com/normation/rudder/batch/AutomaticReportLogger.scala

```diff
@@ -93,114 +96,166 @@ class AutomaticReportLogger(
               }
               updateLastId(id)
+            }
 
-            case eb:EmptyBox =>
-              logger.error("report logger could not fetch latest 100 non compliant reports, retry on next run",eb)
-          }
-
-        case Full(last) =>
-          logger.trace("***** get current log id")
-
-          reportsRepository.getHighestId match {
-            case Full(currentId) if (currentId > last) =>
-              logger.trace("***** log report beetween ids %d and %d".format(last,currentId))
-              reportsRepository.getErrorReportsBeetween(last,currentId,Reportskind) match {
-                case Full(reports) =>
-                  logReports(reports)
-                  updateLastId(currentId)
-                case eb:EmptyBox =>
-                  logger.error("report logger could not fetch latest non compliant reports, try on next run",eb)
-              }
+            isSuccess match {
+              case eb:EmptyBox =>
+                logger.error("report logger could not fetch latest 100 non compliant reports, retry on next run",eb)
+              case Full(x) => //
+            }
 
-            case _ =>
-              logger.trace("***** no reports to log")
-          }
 
-        case Failure(message,_,_) =>
-          logger.error("could not fetch last id, don't log anything, wait next run, cause is %s".format(message))
-      }
+          case Full(lastId) =>
+            logger.trace("***** get current highest non-compliant log id")
+            val highest = reportsRepository.getHighestErrorId(reportskind)
+            logger.trace(s"***** highest non-compliant log id = ${highest} and last processed id = ${lastId}")
+            highest match {
+              case Full(currentId) if (currentId > lastId) =>
+                logReportsBetween(lastId, currentId)
 
-      //schedule next log, every minute
-      LAPinger.schedule(this, StartAutomaticReporting, reportLogInterval*1000L*60)
-      () //ok for the unit value discarded
+              case _ =>
+                logger.trace("***** no reports to log")
+            }
+
+          case Failure(message,_,_) =>
+            logger.error(s"could not fetch last id, don't log anything, wait next run, cause is ${message}")
+        }
+
+        //schedule next log, every minute
+        LAPinger.schedule(this, StartAutomaticReporting, reportLogInterval*1000L*60)
+        () //ok for the unit value discarded
 
       case _ =>
         logger.error("Wrong message received by non compliant reports logger, do nothing")
     }
 
+    private[this] def logReportsBetween(lastId: Long, currentId: Long): Unit = {
+
+      val batchSize = 10000
+      //log from "from". Return the last logged id
+      def log(from: Long, allNodes: Map[NodeId, NodeInfo], rules: Map[RuleId, Rule],  directives: FullActiveTechniqueCategory): Box[Long] = {
+        for {
+          reports <- reportsRepository.getErrorReportsBeetween(from, currentId, batchSize, reportskind)
```

Comment:
you don't have any error handling there, is it expected ?
if the method getErrorReportsBeetween fails, we'll never know it's during the non compliant error reporting